managers/backup-manager/BackupWatcher.js

```diff
@@ -0,0 +1,151 @@
+const eventmesh = require('../../eventmesh');
+const config = require('../../common/config');
+const logger = require('../../common/logger');
+const CONST = require('../../common/constants');
+const bm = require('./');
+const BaseManager = require('../BaseManager');
+const DBManager = require('../../broker/lib/fabrik/DBManager');
+const errors = require('../../common/errors');
+const Conflict = errors.Conflict;
+
+/* jshint nonew:false */
+new DBManager(); //to start the BnRStatusPoller
+
+class DefaultBackupManager extends BaseManager {
+
+  static _processBackup(changeObjectBody) {
+    const changedOptions = JSON.parse(changeObjectBody.spec.options);
+    logger.info('Triggering backup with the following options:', changedOptions);
+    const plan = catalog.getPlan(changedOptions.plan_id);
+    return bm.createManager(plan)
+      .then(manager => manager.startBackup(changedOptions));
+  }
+
+  static _processAbort(changeObjectBody) {
+    const changedOptions = JSON.parse(changeObjectBody.spec.options);
+    return eventmesh.apiServerClient.getOperationOptions({
+        resourceId: changedOptions.instance_guid,
+        operationName: CONST.APISERVER.ANNOTATION_NAMES.BACKUP,
+        operationType: CONST.APISERVER.ANNOTATION_TYPES.BACKUP,
+        operationId: changedOptions.guid
+      })
+      .then(options => {
+        return Promise.try(() => {
+          const plan = catalog.getPlan(options.plan_id);
+          return bm.createManager(plan);
+        }).then(manager => manager.abortLastBackup(options));
+      });
+  }
+
+  static _processDelete(changeObjectBody) {
+    const changedOptions = JSON.parse(changeObjectBody.spec.options);
+    return eventmesh.apiServerClient.getOperationOptions({
+        resourceId: changedOptions.instance_guid,
+        operationName: CONST.APISERVER.ANNOTATION_NAMES.BACKUP,
+        operationType: CONST.APISERVER.ANNOTATION_TYPES.BACKUP,
+        operationId: changedOptions.guid
+      })
+      .then(options => {
+        return Promise.try(() => {
+          const plan = catalog.getPlan(options.plan_id);
+          return bm.createManager(plan);
+        }).then(manager => manager.deleteBackup(options));
+      });
+  }
+
+  /**
+   * @description Registers backup watcher with worker callback
+   */
+  registerWatcher() {
+    logger.info(`Registering Backup watcher`);
+    const queryString = `state in (${CONST.APISERVER.RESOURCE_STATE.IN_QUEUE},${CONST.OPERATION.ABORT},${CONST.APISERVER.RESOURCE_STATE.DELETE})`;
+    return eventmesh.apiServerClient.registerWatcher('backup', 'defaultbackup', this.worker, queryString)
+      .then(stream => {
+        logger.info(`Successfully set watcher with query string:`, queryString);
+        return setTimeout(() => {
+          try {
+            logger.info(`Refreshing stream after ${CONST.APISERVER.WATCHER_REFRESH_INTERVAL}`);
+            stream.abort();
+            return this.registerWatcher();
```

Comment:
You keep registering yourself in recursion? So its not truly a watcher then, it should have been poller? :-)